src/overlays/actors/ovl_En_Tru/z_en_tru.c

```diff
@@ -31,91 +89,1189 @@ const ActorInit En_Tru_InitVars = {
+            if (this->unk_384 > this->unk_384 + 1) {
+                this->unk_384 = this->path->count - 2;
+                ret = true;
+            }
+            this->unk_384++;
+        }
+        Math_Vec3s_ToVec3f(&sp34, &sp4C[this->unk_384]);
+        Math_Vec3f_Copy(&sp40, &this->actor.world.pos);
+        phi_a1 = Math_Vec3f_Yaw(&sp40, &sp34);
+    } else {
+        phi_a1 = this->actor.world.rot.y;
+    }
+
+    Math_ApproachS(&this->actor.world.rot.y, phi_a1, 4, 3640);
+    this->actor.shape.rot.y = this->actor.world.rot.y;
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    if ((s32)(this->actor.floorHeight + 80.0f) >= (s32)this->actor.world.pos.y) {
+        func_80A86770(this);
+    }
+
+    return ret;
+}
+
+s32 func_80A875AC(Actor* thisx, GlobalContext* globalCtx) {
+    EnTru* this = THIS;
+    s32 ret = false;
+
+    switch (this->unk_364) {
+        case 0:
+            if ((this->unk_34E & 0x40) || (gSaveContext.weekEventReg[16] & 0x10)) {
+                this->unk_374 = this->actor.cutscene;
+                this->unk_364++;
+            } else {
+                this->unk_364++;
+                this->unk_364++;
+                break;
+            }
+
+        case 1:
+            if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+                ActorCutscene_Stop(0x7C);
+                ActorCutscene_SetIntentToPlay(this->unk_374);
+            } else if (ActorCutscene_GetCanPlayNext(this->unk_374)) {
+                ActorCutscene_StartAndSetUnkLinkFields(this->unk_374, &this->actor);
+                this->unk_364++;
+            } else {
+                ActorCutscene_SetIntentToPlay(this->unk_374);
+            }
+            break;
+
+        case 2:
+            if ((this->unk_37C != 5) && (this->unk_37C != 6)) {
+                func_80A86924(this, 3);
+                this->unk_364++;
+            } else {
+                func_80A86924(this, 4);
+                ret = true;
+            }
+            break;
+
+        case 3:
+            if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+                this->unk_364++;
+                func_80A86924(this, 4);
+                ret = true;
+            }
+            break;
+    }
+
+    if (ret == true) {
+        if (this->unk_390 != 0) {
+            this->unk_34E |= 8;
+            this->unk_378 = func_80A87880;
+        } else {
+            this->unk_378 = func_80A8777C;
+        }
+        this->unk_364 = 0;
+    }
+
+    return ret;
+}
+
+s32 func_80A8777C(Actor* thisx, GlobalContext* globalCtx) {
+    s32 temp_v0;
+    s32 ret = 0;
+
+    temp_v0 = func_80152498(&globalCtx->msgCtx);
+
+    switch (temp_v0) {
+        default:
+            if (temp_v0 != 0x10) {
+                break;
+            }
+            goto label;
+
+        case 4:
+        case 5:
+            if (!func_80147624(globalCtx)) {
+                break;
+            }
+        label:
+            temp_v0 = func_80123810(globalCtx);
```

Comment:
I would prefer the wrapped in an if rather than a goto, but if everyone else hates that, I'm prepared to yield.

Reply:
I changed it, but I think it looks way less clear and is ahrder to see and udnerstand on looking at the function.